core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/signing/TransactionSerializer.java

```diff
@@ -33,23 +38,50 @@ public TransactionSerializer(final ArtifactSigner signer, final long chainId) {
   }
 
   public String serialize(final Transaction transaction) {
-    final byte[] bytesToSign = transaction.rlpEncode(chainId);
+    if (transaction.isEip1559()) {
```

Comment:
Might be nice to have a unit test showing we encode serialize transactions correctly

Reply:
true, this is only partially covered by EthTransactionTest (the rlpEncode bit) and The AT. Will consider for next time